File: client/src/types/rating.ts
export interface Rating {
  ratingId: string;
  account: {
    accountId: string;
    accountName: string;
    accountPhone?: string;
  };
  product: {
    productId: string;
    productName: string;
  };
  orderItemId?: string; // Link to specific order item
  comment: string;
  ratingStar: number;
  status: number;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateRatingRequest {
  accountId: string;
  productId: string;
  orderItemId?: string; // Link rating to specific order item
  comment: string;
  ratingStar: number;
}

export interface UpdateRatingRequest {
  comment: string;
  status: number;
  ratingStar: number;
}

File: microservice/review-service/src/main/java/fruitshop/review_service/entity/Rating.java
package fruitshop.review_service.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "ratings")
@Data
public class Rating {
    @Id
    private String ratingId;

    @Column(nullable = false)
    private String accountId;

    @Column(nullable = false)
    private String productId;

    private String orderItemId;

    private String comment;
    private double ratingStar;
    private int status;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    public void generateIdIfAbsent() {
        if (this.ratingId == null) {
            this.ratingId = UUID.randomUUID().toString();
        }
    }
}
